lib/segment/src/index/hnsw_index/graph_layers_builder.rs

```diff
@@ -714,7 +728,11 @@ mod tests {
             let links_orig = &graph_layers_orig.links.links_vec(idx as PointOffsetType, 0);
             let links_builder = graph_layers_builder.links_layers[idx][0].read();
             let link_container_from_builder = links_builder.iter().copied().collect::<Vec<_>>();
-            assert_eq!(links_orig, &link_container_from_builder);
+            let m = if compressed { M * 2 } else { 0 };
```

Comment:
As mentioned in another PR, logic `max_m = m * 2` should have one source of true. In means moving in a separate function